### ISSUE TITLE
IllegalAnnotationExceptions Two classes have the same XML type name

### ISSUE THREAD

**Issue by ppalaga** [quarkiverse/quarkus-cxf#730]
This happens with the current Quarkus main branch. Originaly reported in Camel Quarkus: https://github.com/apache/camel-quarkus/issues/4603

```
[INFO] Running io.quarkiverse.cxf.it.ws.securitypolicy.server.CxfWssSecurityPolicyServerTest
[ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 4, Time elapsed: 2.727 s <<< FAILURE! - in io.quarkiverse.cxf.it.ws.securitypolicy.server.CxfWssSecurityPolicyServerTest
[ERROR] wsdl  Time elapsed: 0.05 s  <<< ERROR!
java.lang.RuntimeException: 
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.jaxb.deployment.JaxbProcessor#setupJaxbContextConfig threw an exception: java.lang.IllegalStateException: Failed to configure JAXB context
        at io.quarkus.jaxb.deployment.JaxbProcessor.validateContext(JaxbProcessor.java:392)
        at io.quarkus.jaxb.deployment.JaxbProcessor.setupJaxbContextConfig(JaxbProcessor.java:317)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:1589)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: org.glassfish.jaxb.runtime.v2.runtime.IllegalAnnotationsException: 47 counts of IllegalAnnotationExceptions
Two classes have the same XML type name "{http://www.w3.org/2000/09/xmldsig#}DSAKeyValueType". Use @XmlType.name and @XmlType.namespace to assign different names to them.
        this problem is related to the following location:
                at org.apache.xml.security.binding.xmldsig.DSAKeyValueType
                at public org.apache.xml.security.binding.xmldsig.DSAKeyValueType org.apache.xml.security.binding.xmldsig.ObjectFactory.createDSAKeyValueType()
                at org.apache.xml.security.binding.xmldsig.ObjectFactory
                at protected java.util.List org.apache.xml.security.binding.xmldsig.TransformType.content
                at org.apache.xml.security.binding.xmldsig.TransformType
                at protected java.util.List org.apache.xml.security.binding.xmlenc.TransformsType.transform
                at org.apache.xml.security.binding.xmlenc.TransformsType
                at public org.apache.xml.security.binding.xmlenc.TransformsType org.apache.xml.security.binding.xmlenc.ObjectFactory.createTransformsType()
                at org.apache.xml.security.binding.xmlenc.ObjectFactory
                at protected java.util.List org.apache.xml.security.binding.xmlenc.EncryptionMethodType.content
                at org.apache.xml.security.binding.xmlenc.EncryptionMethodType
        this problem is related to the following location:
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.DSAKeyValueType
Two classes have the same XML type name "{http://www.w3.org/2000/09/xmldsig#}ObjectType". Use @XmlType.name and @XmlType.namespace to assign different names to them.
        this problem is related to the following location:
                at org.apache.xml.security.binding.xmldsig.ObjectType
                at public jakarta.xml.bind.JAXBElement org.apache.xml.security.binding.xmldsig.ObjectFactory.createObject(org.apache.xml.security.binding.xmldsig.ObjectType)
                at org.apache.xml.security.binding.xmldsig.ObjectFactory
                at protected java.util.List org.apache.xml.security.binding.xmldsig.TransformType.content
                at org.apache.xml.security.binding.xmldsig.TransformType
                at protected java.util.List org.apache.xml.security.binding.xmlenc.TransformsType.transform
                at org.apache.xml.security.binding.xmlenc.TransformsType
                at public org.apache.xml.security.binding.xmlenc.TransformsType org.apache.xml.security.binding.xmlenc.ObjectFactory.createTransformsType()
                at org.apache.xml.security.binding.xmlenc.ObjectFactory
                at protected java.util.List org.apache.xml.security.binding.xmlenc.EncryptionMethodType.content
                at org.apache.xml.security.binding.xmlenc.EncryptionMethodType
        this problem is related to the following location:
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectType
                at public jakarta.xml.bind.JAXBElement org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectFactory.createObject(org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectType)
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectFactory
                at protected java.util.List org.apache.cxf.ws.security.sts.provider.model.xmldsig.CanonicalizationMethodType.content
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.CanonicalizationMethodType
                at protected org.apache.cxf.ws.security.sts.provider.model.xmldsig.CanonicalizationMethodType org.apache.cxf.ws.security.sts.provider.model.xmldsig.SignedInfoType.canonicalizationMethod
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.SignedInfoType
                at protected org.apache.cxf.ws.security.sts.provider.model.xmldsig.SignedInfoType org.apache.cxf.ws.security.sts.provider.model.xmldsig.SignatureType.signedInfo
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.SignatureType
The element name {http://www.w3.org/2000/09/xmldsig#}Object has more than one mapping.
        this problem is related to the following location:
                at public jakarta.xml.bind.JAXBElement org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectFactory.createObject(org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectType)
                at org.apache.cxf.ws.security.sts.provider.model.xmldsig.ObjectFactory
                at protected java.util.List org.apache.cxf.ws.security.sts.provider.model.xmldsig.CanonicalizationMethodType.content
```

**Comment by ppalaga** [comment]
It looks like CXF 4 does not depend on santuario:xmlsec anymore. So quarkus-cxf should also stop depending on Qverse xmlsec.

**Comment by ppalaga** [comment]
> It looks like CXF 4 does not depend on santuario:xmlsec anymore. So quarkus-cxf should also stop depending on Qverse xmlsec.

No, this is not true. The santuario:xmlsec dependecy is still there, although now it is transitive.

**Comment by ppalaga** [comment]
Duplicate JAXB entities seem to be a normal situation in CXF. I do not think we can avoid those by excluding artifacts or classes. I think Quarkus JAXB should offer a way to avoid validating the default JAXB context. I have filed https://github.com/quarkusio/quarkus/issues/31646

**Comment by ppalaga** [comment]
The fix https://github.com/quarkusio/quarkus/pull/31666 was merged in Quarkus main branch and should reach us with Quarkus 3.0.0.Alpha5+1

**Comment by ppalaga** [comment]
The issue was fixed in Quarkus 3.0.0.Alpha6 and our workaround was reverted via 5e483429608974abdc7bffc2f281a8e683711abe
